### PR TITLE
Fix slot 0xFF system crash issue.

### PR DESCRIPTION
**Pull request by jyao1**
Fix: https://github.com/DMTF/libspdm/issues/1541

Signed-off-by: Jiewen Yao <jiewen.yao@intel.com>

Also need spdm-emu: https://github.com/DMTF/spdm-emu/pull/194

### CODE REVIEW

**Review by steven-bellock — APPROVED** [review]

File: library/spdm_responder_lib/libspdm_rsp_key_exchange.c [hunk 1]
```diff
@@ -362,10 +362,16 @@ libspdm_return_t libspdm_get_response_key_exchange(void *context,
...
-        libspdm_init_mut_auth_encap_state(
-            context, spdm_response->mut_auth_requested);
-        spdm_response->req_slot_id_param =
-            (spdm_context->encap_context.req_slot_id & 0xF);
+        if (spdm_response->mut_auth_requested == SPDM_KEY_EXCHANGE_RESPONSE_MUT_AUTH_REQUESTED) {
+            /* no need to libspdm_init_mut_auth_encap_state() because of no ENCAP message */
+            spdm_response->req_slot_id_param =
+                (spdm_context->encap_context.req_slot_id & 0xF);
+        } else {
+            libspdm_init_mut_auth_encap_state(
```
Comment (xiangfeima) [inline library/spdm_responder_lib/libspdm_rsp_key_exchange.c:370]: Would you please confirm if this should be updated as well?
As far as I can see, the possible sequences should be only
`     *    2.1 GET_DIGEST/GET_CERTIFICATE (spdm_context->encap_context.req_slot_id != 0xFF) */`
`     *    2.2 N/A (SPDM_GET_CAPABILITIES_REQUEST_FLAGS_PUB_KEY_ID_CAP, spdm_context->encap_context.req_slot_id must be 0xFF)`

https://github.com/DMTF/libspdm/blob/befd3a8c41427feb28719db4324ec2f82ed62790/library/spdm_responder_lib/libspdm_rsp_encap_response.c#L199-L226

Comment (jyao1) [reply library/spdm_responder_lib/libspdm_rsp_key_exchange.c:370]: you are right. Will update as well.

Comment (jyao1) [reply library/spdm_responder_lib/libspdm_rsp_key_exchange.c]: @xiangfeima, please review the update if possible.

File: library/spdm_responder_lib/libspdm_rsp_encap_response.c [hunk 2]
```diff
@@ -198,30 +196,47 @@ void libspdm_init_mut_auth_encap_state(libspdm_context_t *spdm_context, uint8_t
 
     /* Possible Sequence:
      * 2. Session Mutual Auth: (spdm_context->last_spdm_request_session_id_valid)
-     *    2.1 GET_DIGEST/GET_CERTIFICATE (spdm_context->encap_context.req_slot_id != 0xFF)
-     *    2.2 GET_DIGEST (spdm_context->encap_context.req_slot_id == 0xFF)
-     *    2.3 N/A (SPDM_GET_CAPABILITIES_REQUEST_FLAGS_PUB_KEY_ID_CAP)*/
+     *    2.1 GET_DIGEST/GET_CERTIFICATE (MUT_AUTH_REQUESTED_WITH_ENCAP_REQUEST, encap_context.req_slot_id must not be 0xFF)
+     *    2.2 GET_DIGEST (MUT_AUTH_REQUESTED_WITH_GET_DIGESTS, encap_context.req_slot_id must not be 0xFF)
```
Comment (xiangfeima) [inline library/spdm_responder_lib/libspdm_rsp_encap_response.c:200]: Hi @jyao1 , I don't see how this 2.2 sequence comes here.
According to SPDM specification v1.2, section 10.22.2 Optimized encapsulated request flow, paragraph 610, the `Optimized encapsulated request flow` (case `SPDM_KEY_EXCHANGE_RESPONSE_MUT_AUTH_REQUESTED_WITH_GET_DIGESTS`) just makes the Requester be able to skip the initial `GET_ENCAPSULATED_REQUEST` to get encapsulated `GET_DIGESTS` request from the Responder. And the encapsulated `GET_CERTIFICATE` flow afterwards may still needed, as the Figure 22 — Optimized session-based mutual authentication example shows.

Though the SDPM specification dosen't have clear description about that encapsulated `GET_CERTIFICATE` flow can be skipped in mutual authentication process if encapsulated `GET_DIGESTS` result is good, just as same as the one-way authentication flow does, I'm just guessing the idea here is to handle such scenario. However I think we would never know it at here at this moment but after the encapsulated `DIGESTS` response is processed by the Responder.

Comment (jyao1) [reply library/spdm_responder_lib/libspdm_rsp_encap_response.c]: you are right.  my mistake. Flow is updated.

**Comment by xiangfeima** [comment]
Looks all good now 👍

**Pull request merged**